service/src/test/java/com/ericsson/eiffel/remrem/generate/service/EiffelTemplateGenerateTest.java

```diff
@@ -0,0 +1,112 @@
+public class EiffelTemplateGenerateTest {
+
+    @Autowired
+    private MsgService msgServices;
+
+    private Logger log = (Logger) LoggerFactory.getLogger(EiffelTemplateGenerateTest.class);
+
+    @Test
+    public void testEiffelSemanticsTemplate() throws IOException {
+        for (EiffelEventType enumValue:EiffelEventType.values()
+        ) {
+            testEventType(enumValue.getEventName());
+
+        }
+    }
+
+    /**
+     * In this basically we generate event and schemas version and compare it
+     * @param eventName name of the event
+     */
+    protected void testEventType(String eventName){
+
+        JsonElement jsonTemplate= msgServices.getEventTemplate(eventName);
+        JsonObject jsonObjectTemplate = jsonTemplate.getAsJsonObject();
+
+        String templateVersion = jsonObjectTemplate.getAsJsonObject("msgParams")
+                .getAsJsonObject("meta").get("version").getAsString();
+
+
+        EiffelEventType eiffelType = EiffelEventType.fromString(eventName);
+        EiffelValidator validator = EiffelOutputValidatorFactory.getEiffelValidator(eiffelType);
+
+        String schemaResourceName = null;
+        try {
+            // Note, that this isn't a standard solution. Unfortunately, there's no standard way how to access schema
+            // resource name from validator. That's why Java reflection takes place. It's acceptable just because it's
+            // a takes case, and it isn't part of production code.
+            // Get access to EiffelValidator.schemaResourceName even it's declared as private.
+
+            Field schemaResourceNameField = EiffelValidator.class.getDeclaredField("schemaResourceName");
+            schemaResourceNameField.setAccessible(true);
+            schemaResourceName = (String)schemaResourceNameField.get(validator);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            assertNull("cannot get field name", e);
+            throw new RuntimeException(e);
+        }
+        assertNotNull("Cannot get the schema file name", schemaResourceName);
+
+        ClassLoader classLoader = validator.getClass().getClassLoader();
+        InputStream inputStream = classLoader.getResourceAsStream(schemaResourceName);
+        String stringTypeSchemas = null;
+        try {
+            stringTypeSchemas = new String(inputStream.readAllBytes());
+        } catch (IOException e) {
+            fail("Cannot read resource " + schemaResourceName);
+        }
+
+        JsonObject SchemasAsJsonObject = JsonParser.parseString(stringTypeSchemas).getAsJsonObject();
+        String schemasVersion = SchemasAsJsonObject.getAsJsonObject("properties").getAsJsonObject("meta").
+                getAsJsonObject("properties").getAsJsonObject("version").
+                get("default").getAsString();
+
+
+        // When using assert(...) if a logical test fails the unit test is aborted and the rest of the unit test isn't run.
```

Comment:
The comment doesn't explain why standard assert isn't used. Please, use comment below and put it inside the 'if' statement:

At the moment only semantics library 2.2.1 has been hot-fixed. The later versions still have the issue (it hasn't been decided how/when/if the issue will be resolved). If standard "assert" was used the test would fail for the other versions of the library, and this is unwanted as the test would fail as a whole.